### PR TITLE
Equation Handling for Formal Elements and Xrefs

### PR DESCRIPTION
**Pull request by christopappas**

### CODE REVIEW

File: htmlbook-xsl/xspec/common.xspec [hunk 1]
```diff
@@ -353,6 +353,234 @@ sect5:none
...
+    	
+    	<x:scenario label="With label.formal.with.ancestor disabled">
+    		<x:context select="h:section[@data-type='chapter']//h:div[@data-type='equation'][3]"   mode="label.markup">
+    			<x:param name="label.formal.with.ancestor" select="0"/>
+    		</x:context>
+    		<x:expect label="The appropriate numeration value should be generated" >2</x:expect>
+    		
+    	</x:scenario>
+    	
+    	<x:scenario label="With label.formal.with.ancestor disabled">
```
Comment (sandersk) [inline htmlbook-xsl/xspec/common.xspec]: Hey Chris, 

Minor note on the next three scenarios. I think the `label` value for these should be "With label.formal.with.ancestor enabled"? Actual test logic looks great.

File: htmlbook-xsl/xspec/common.xspec [hunk 2]
```diff
@@ -353,6 +353,234 @@ sect5:none
...
+	</x:scenario>
+	
+	<x:scenario label="With label.formal.with.ancestor disabled">
+		<x:context select="h:section[@data-type='preface']//h:div[@data-type='equation'][3]"   mode="label.markup">
+			<x:param name="label.formal.with.ancestor" select="0"/>
+		</x:context>
+		<x:expect label="The appropriate numeration value should be generated" >2</x:expect>
+		
+	</x:scenario>
+
```
Comment (sandersk) [inline htmlbook-xsl/xspec/common.xspec]: Hey Chris,

Same note here. For the next three scenarios. I think the label value for these should be "With label.formal.with.ancestor enabled"?

File: htmlbook-xsl/xspec/common.xspec [hunk 3]
```diff
@@ -353,6 +353,234 @@ sect5:none
...
+	</x:scenario>
+	
+	<x:scenario label="With label.formal.with.ancestor disabled">
+		<x:context select="h:section[@data-type='appendix']//h:div[@data-type='equation'][3]"   mode="label.markup">
+			<x:param name="label.formal.with.ancestor" select="0"/>
+		</x:context>
+		<x:expect label="The appropriate numeration value should be generated" >2</x:expect>
+		
+	</x:scenario>
+
```
Comment (sandersk) [inline htmlbook-xsl/xspec/common.xspec]: Hey Chris,

Same note here. For the next three scenarios. I think the label value for these should be "With label.formal.with.ancestor enabled"?

File: htmlbook-xsl/xspec/common.xspec (hunk 1 above)
Comment (christopappas) [inline htmlbook-xsl/xspec/common.xspec]: Ack, good catch. Looks like this is the case for the next set of tests too. I'll double check all of these once more and update the PR from staging.

Sorry about that!

**Comment by christopappas** [comment]
Ok @sandersk, I've made the suggested changed to the test labels in common.xspec in the staging branch.

File: htmlbook-xsl/xspec/xrefgen.xspec [hunk 4]
```diff
@@ -265,6 +265,80 @@ toc:lower-roman
...
+      <section data-type="chapter">
+      <div data-type="equation">
+        <p data-type="tex">$2 + 2 = 4$</p>
+      </div>
+      </section>
+    </x:context>
+    <x:expect label="Gentext should be created for the xref">???</x:expect>
+  </x:scenario>
+  
+  <x:scenario label="When an XREF points to a formal equation with a caption">
```
Comment (sandersk) [inline htmlbook-xsl/xspec/xrefgen.xspec]: I think the above label should be "without a caption"

File: htmlbook-xsl/xspec/xrefgen.xspec [hunk 5]
```diff
@@ -265,6 +265,80 @@ toc:lower-roman
...
+          <p data-type="tex">$2 + 2 = 4$</p>
+        </div>
+      </section>
+    </x:context>
+    <x:expect label="Gentext should be created for the xref">Equation 1-1</x:expect>
+  </x:scenario>
+  
+<!-- Equation in a preface -->
+  <x:scenario label="When an XREF points to a formal equation with a caption">
+    <x:context mode="xref-to" select="(//h:div[@data-type='equation'])[1]">
```
Comment (sandersk) [inline htmlbook-xsl/xspec/xrefgen.xspec]: I think the above label should be "without a caption"

File: htmlbook-xsl/xspec/xrefgen.xspec [hunk 6]
```diff
@@ -265,6 +265,80 @@ toc:lower-roman
...
+          <h5/>
+          <p data-type="tex">$2 + 2 = 4$</p>
+        </div>
+      </section>
+    </x:context>
+    <x:expect label="Gentext should be created for the xref">Equation P-1</x:expect>
+  </x:scenario>
+  
+ <!-- Equation in an appendix -->
+  <x:scenario label="When an XREF points to a formal equation with a caption">
```
Comment (sandersk) [inline htmlbook-xsl/xspec/xrefgen.xspec]: I think the above label should be "without a caption"?

**Comment by christopappas** [comment]
Ok @sandersk , I've made the changes to xrefgen.xspec. Sorry for the shoddy labels!

**Comment by sandersk** [comment]
Awesome, thanks, @christopappas. Merging!

**Pull request merged**